Yubico.YubiKey/src/Yubico/YubiKey/Piv/PivSession.Crypto.cs

```diff
@@ -155,6 +155,7 @@ public byte[] Sign(byte slotNumber, ReadOnlyMemory<byte> dataToSign)
             // This will verify the slot number and dataToSign length. If one or
             // both are incorrect, the call will throw an exception.
             var signCommand = new AuthenticateSignCommand(dataToSign, slotNumber);
+            _yubiKeyDevice.ThrowIfUnsupportedAlgorithm(signCommand.Algorithm);
```

Comment:
This call might throw `NotSupportedException` which is not mentioned in the function's documentation above.

Reply:
got it, adding a documentation ref